### PR TITLE
remove $ from class name

### PR DESCRIPTION
**Pull request by pjfanning**
* relates to #491

### CODE REVIEW

File: remote/src/main/scala/org/apache/pekko/remote/transport/PekkoPduCodec.scala
Comment (He-Pin) [inline remote/src/main/scala/org/apache/pekko/remote/transport/PekkoPduCodec.scala:131]: Hum

**Comment by mdedetrich** [comment]
So MIMA is breaking which is kind of expected, otherwise it looks good (will have to set MIMA to 1.0.1)

**Comment by mdedetrich** [comment]
@pjfanning Regarding the MIMA changes, I would actually advocating to disable MIMA and/or ignore the CI rather than adding more filters because if someone breaks this again in the future then MIMA won't catch it.

I think the most practical would be to see if CI works with the newly added MIMA filters and then remove the filters just before merge

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Blocking PR due to https://github.com/apache/incubator-pekko/pull/492#issuecomment-1638142692, will unblock when the MIMA filters get removed.

File: remote/src/main/scala/org/apache/pekko/remote/transport/PekkoPduCodec.scala
Comment (He-Pin) [reply remote/src/main/scala/org/apache/pekko/remote/transport/PekkoPduCodec.scala:131]: image

Comment (He-Pin) [reply remote/src/main/scala/org/apache/pekko/remote/transport/PekkoPduCodec.scala:131]: Seems IDEA messed this.

**Comment by pjfanning** [comment]
> Blocking PR due to [#492 (comment)](https://github.com/apache/incubator-pekko/pull/492#issuecomment-1638142692), will unblock when the MIMA filters get removed.

Why do we need to remove the mima filters? There is no 1.0.1 release to compare with.

Are you advocating that we should turn off MIMA checking altogether?

**Comment by pjfanning** [comment]
> @pjfanning Do you plan to have this get merged and cherry-pick a pr against the 1.0.x branch?

If you check the mailing list, @mdedetrich wants us to delay using the `1.0.x` branch.

And the mailing list also includes a suggested patching process that involves merging to `main` first and backporting selected PRs to `1.0.x` as opposed to merging to `1.0.x` first.

**Comment by pjfanning** [comment]
> @pjfanning Regarding the MIMA changes, I would actually advocate to disable MIMA and/or ignore the MIMA CI check rather than adding more filters because if someone breaks this again in the future then MIMA won't catch it.
> 
> I think the most practical would be to see if CI works with the newly added MIMA filters and then remove the filters just before merge and then after this PR is merged set MIMA to work on Pekko 1.0.1.

Can't we rebase the mima checks to v1.0.1 after it is released and remove the mima filters in this PR as part of that.

**Comment by mdedetrich** [comment]
> Pekko HTTP does not use pekko-remote (where the issue is). It's not
very tidy but only Pekko HTTP users who also need Pekko Remote and
Clustering features would need to upgrade to Pekko (Core) 1.0.1.

As long as we don't forget, sure. I will review the PR by end of tonight (want to also check other sections of code to make sure we catch everything).

**Review by mdedetrich — APPROVED** [review]
lgtm, but lets make sure that we don't merge anything into main unless necessary (which should be a given considering that we are making a new release which will create a code freeze) and also undo the MIMA checks that this PR adds since the filters that are being added are covering a public API.

**Comment by pjfanning** [comment]
The mima filters are for PekkoPduProtobufCodec - a package private object. There is no public API affected by this change or its mima filters.

**Pull request merged**
